resolwe/flow/models/utils/duplicate.py

```diff
@@ -0,0 +1,301 @@
+"""Resolwe models duplicate utils."""
+from copy import deepcopy
+
+from django.contrib.auth.models import Permission
+from django.contrib.contenttypes.models import ContentType
+from django.core.exceptions import ValidationError
+from django.db import IntegrityError, models, transaction
+from django.utils import timezone
+from guardian.models import GroupObjectPermission, UserObjectPermission
+
+from resolwe.flow.utils import iterate_fields
+
+
+def _check_permissions(objects, permission, user):
+    """"""
+    for obj in objects:
+        if not user.has_perm(permission, obj):
+            raise ValidationError(
+                "User doesn't have '{}' permission on {}.".format(permission, obj)
+            )
+
+
+def _bulk_copy_permissions(content_type, pk_mapping):
+    for permission_cls in [GroupObjectPermission, UserObjectPermission]:
+
+        permissions_qs = permission_cls.objects.filter(
+            content_type=content_type, object_pk__in=pk_mapping.keys()
+        )
+        new_permissions = []
+
+        for permission in permissions_qs:
+            permission.pk = None
+            permission.object_pk = pk_mapping[int(permission.object_pk)]
+            new_permissions.append(permission)
+        permission_cls.objects.bulk_create(new_permissions)
+
+
+def _bulk_assign_contributor_permission(content_type, pks, user):
+    permissions = Permission.objects.filter(content_type=content_type)
+
+    new_permissions = []
+    for permission in permissions:
+        for pk in pks:
+            new_permissions.append(
+                UserObjectPermission(
+                    content_type=content_type,
+                    object_pk=pk,
+                    permission=permission,
+                    user=user,
+                )
+            )
+    UserObjectPermission.objects.bulk_create(new_permissions, ignore_conflicts=True)
+
+
+def _rewire_foreign_key(objects, target, pk_mapping):
+    """"""
+    if not objects:
+        return
+
+    # Check that all objects are instances of the same class.
+    obj_class = {obj.__class__ for obj in objects}
+    if len(obj_class) > 1:
+        print(obj_class)
+        raise ValueError("All objects must be instances of the same class.")
+    obj_class = obj_class.pop()
+
+    # Find fields pointing to the target model.
+    to_map = []
+    for field in obj_class._meta.fields:
+        if type(field) == models.ForeignKey and field.related_model == target:
+            to_map.append(field.name)
+
+    # Rewire references.
+    new_objects = []
+    for obj in objects:
+        new_obj = deepcopy(obj)
+        for field_name in to_map:
+            field_name = "{}_id".format(field_name)
+            value = getattr(obj, field_name)
+            if not value or value not in pk_mapping:
```

Comment:
I would use 

```
if value and value in pk_mapping:
    setattr(...)  
```
Or just `if value in pk_mapping` since None is probably not contained in the mapping.